### DISCUSSION TITLE
Font and Text Management

### DISCUSSION THREAD

**Discussion by bozvary** [felixmariotto/three-mesh-ui#258] [General]
Hi There, 

I would have two questions, and I am still looking for answers or solutions.

1. 
	Is there any way to use 
		a, HTML sup tag, e.g. TM mark, x2, cm2, m3, etc...
		b, HTML sub tag H2O

2. 
	Can we use RTL languages?

Thanks in advance for helping me in the right direction

**Comment by swingingtom** [comment]
Hi @bozvary,
1.a Not possible to use tags. However, you can embed those characters in the msdf charset. If you inspect the following `cm²` `m³` you won't see any sup tag. Please note not all fonts provide those characters.

When exporting msdf (see wiki) the sup2 and sup3 characters have been requested in the charset.
![image](https://user-images.githubusercontent.com/17314210/229562752-9faba8f2-3708-4ef4-8231-aa0105067a75.png)

Here is a simple demo https://codesandbox.io/s/sup-square-jd414w?file=/src/sandbox.js

1.b No. Same as 1.a could work here too. 
2. No, this isn't possible right now. 

For the question 1, if you own rights, you can use a font editor such as fontforge to put your own characters in the font file if those don't isset yet.

**Reply by bozvary** [reply]
Thank you @swingingtom for the quick reply. 
I was thinking of the 1, solution, but I wanted something dynamic, the "x2" and "m3" was just an example but can be any other 'sup' or 'sub' e.g any text or word.
I have a web experience with 360 panoramas, where we show hotspots and different content in popups. I am implementing this experience into VR. So, when you edit the web experience content, you can see it in VR as well, and the text in the popup can be any text that it's created for the web. So it's completely dynamic, and I cannot create a font for each popup content, that can be changed anytime. What was in my mind, was to approach "somehow" with "inline blocks". So instead of one "textbook", I start splitting the text and if I find anywhere a "sub" or "sup" I add a new "InlineBlock" and continue with the normal "textblock". Just somehow needs to manage the position "bottom/up" and the font size to be smaller.
Any suggestion on how would be best to do it?

2, Is it any plan in the future to implement RTL?

Thank you so much

**Reply by swingingtom** [reply]
1. Understood. That was clearly just a workaround that would have help. Once your web application would have its design defined, you could have duplicated the font to only sub or sub character. Then mixing fonts on Inline elements : 
```javascript
myBlock.Add( 
    new Text({fontFamily:"regular", content:"cm"}),
    new Text({fontFamily: "sup", content: "square"}) 
);
```

Your suggestion seems interesting, If you manage to produce a sandbox I would be glad to have a review. Just keep in mind InlineBlock are used to display kind of image/emojis inside a Inline layout (sequence of chars).

Not sure how it would end up, but on v7, margin and padding have 4 side properties and can be applied to span. I might produce a sandbox if I have some time.

2. No. Not really. I was thinking of implementing it when it would be required for one of my projects. Or relying on some contributors with project requiring it. Whereas it should'nt be that hard, it would have to fit with other features such as alignment, whitespace, etc... So it might be delicate

**Reply by bozvary** [reply]
Thank you, @swingingtom,

I am a bit busy this week, but I am trying to manage something with these fonts, 
`myBlock.Add( 
    new Text({fontFamily:"regular", content:"cm"}),
    new Text({fontFamily: "sup", content: "square"}) 
);`

I haven't thought about that (Sofar, I managed the font on the Block level and not on the Text level; Hopefully, with this approach, I will be able to manage what I need. Once it's done, I will attach an example.


2, if I can manage at some point the RTL, I will share my solutions
